programs/launcher/main.cpp

```diff
@@ -402,19 +434,53 @@ launcher_def::initialize (const variables_map &vmap) {
   alias_base = "testnet_";
   next_node = 0;
 
+  load_servers ();
+
   if (prod_nodes > producers)
     prod_nodes = producers;
   if (prod_nodes > total_nodes)
     total_nodes = prod_nodes;
 
-  if (host_map.bindings.empty()) {
-    define_local ();
+  if (bindings.empty()) {
+    define_network ();
   }
 }
 
+void
+launcher_def::load_servers () {
+  if (!server_ident_file.empty()) {
+    try {
+      fc::json::from_file(server_ident_file).as<server_identities>(servers);
+      size_t nodes = 0;
+      for (auto &s : servers.producer) {
+        nodes += s.instances;
+      }
+      prod_nodes = nodes;
+      nodes = 0;
+      for (auto &s : servers.observer) {
+        nodes += s.instances;
+      }
+
+      total_nodes = prod_nodes + nodes;
+      per_host = 1;
+      network.ssh_helper = servers.ssh;
+    }
+    catch (const exception &ex) {
+      elog( "caught assert on fetch_block_by_id, ${ex}",("ex",ex.what()));
+      exit (-1);
+      }
+    catch (...) {
+      cerr << "unable to load server identity file " << server_ident_file << endl;
```

Comment:
Use elog

Reply:
For the same reason stated above, I'd like to defer refactoring these error messages until later.